apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -275,6 +270,14 @@ function main() {
 
     $scope.hocCategoryName = i18n.hoc_category_name;
 
+    // Angular does not offer a reliable way to wait for the template to load,
+    // so do it using a custom event here.
+    $scope.$on('section-page-rendered', () => {
```

Comment:
Good point. I was copying the pattern I found here https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/sites/code.org/pages/public/teacher-dashboard/index.js#L745 . I may explore other options in a future PR.